torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""NLCD dataset."""
+
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class NLCD(RasterDataset):
+    """National Land Cover Database (NLCD) dataset.
+
+    The `NLCD dataset
+    <https://www.usgs.gov/centers/eros/science/national-land-cover-database>`_
+    is a land cover product that covers the United States and Puerto Rico. It is
+    a joint effort between the United States Geological Survey
+    (`USGS <https://www.usgs.gov/>`_) and the Multi-Resolution Land Characteristics
+    Consortium (`MRLC <https://www.mrlc.gov/>`_) which released the first product
+    in 2001 with new updates every five years since then.
+
+    The dataset contains the following 15 classes:
+
+    #. Open Water
+    #. Pereniial Ice/Snow
+    #. Developed, Open Space
+    #. Developed, Low Intensity
+    #. Developed, Medium Intensity
+    #. Developed, High Intensity
+    #. Barren Land (Rock/Sand/Clay)
+    #. Deciduous Forest
+    #. Evergreen Forest
+    #. Mixed Forest
+    #. Shruf/Scrub
+    #. Grassland/Herbaceous
+    #. Pasture/Hay
+    #. Cultivated Crops
+    #. Woody Wetlands
+    #. Emergent Herbaceous Wetlands
+
+    Detailed descriptions of the classes can be found
+    `on this website <https://www.mrlc.gov/data/legends/national-land-cover-database-class-legend-and-description>`_.
+
+    Dataset format:
+
+    * single channel .img file with integer class labels
+
+    If you use this dataset in your research, please use the corresponding citation:
+
+    * 2001: https://doi.org/10.5066/P9MZGHLF
+    * 2006: https://doi.org/10.5066/P9HBR9V3
+    * 2011: https://doi.org/10.5066/P97S2IID
+    * 2016: https://doi.org/10.5066/P96HHBIE
+    * 2019: https://doi.org/10.5066/P9KZCM54
+
+    .. versionadded:: 0.5
+    """  # noqa: E501
+
+    filename_glob = "nlcd_*_land_cover_l48_20210604.img"
+    filename_regex = (
+        r"nlcd_(?P<date>\d{4})_land_cover_l48_(?P<publication_date>\d{8})\.img"
+    )
+    zipfile_glob = "nlcd_*_land_cover_l48_20210604.zip"
+    date_format = "%Y"
+    is_image = False
+
+    url = "https://s3-us-west-2.amazonaws.com/mrlc/nlcd_{}_land_cover_l48_20210604.zip"
+
+    md5s = {
+        2001: "538166a4d783204764e3df3b221fc4cd",
+        2006: "67454e7874a00294adb9442374d0c309",
+        2011: "ea524c835d173658eeb6fa3c8e6b917b",
+        2016: "452726f6e3bd3f70d8ca2476723d238a",
+        2019: "82851c3f8105763b01c83b4a9e6f3961",
+    }
+
+    ordinal_label_map = {
```

Comment:
Can you modify `ChesapeakeCVPR` to import and use this attribute so we don't have to duplicate it?

Reply:
Sorry, I am not sure I understand what you mean here.

Reply:
It looks like we aren't currently mapping NLCD? Let's do this in a separate PR.